beacon_node/http_api/src/lib.rs

```diff
@@ -1631,49 +1712,63 @@ pub fn serve<T: BeaconChainTypes>(
         .and(network_tx_filter.clone())
         .and_then(
             |chain: Arc<BeaconChain<T>>,
-             aggregate: SignedAggregateAndProof<T::EthSpec>,
+             aggregates: Vec<SignedAggregateAndProof<T::EthSpec>>,
              network_tx: UnboundedSender<NetworkMessage<T::EthSpec>>| {
                 blocking_json_task(move || {
-                    let aggregate =
+                    let mut verified_aggregates = Vec::new();
+
+                    // Verify that all messages in the post are valid before processing further
+                    for aggregate in aggregates.as_slice() {
                         match chain.verify_aggregated_attestation_for_gossip(aggregate.clone()) {
-                            Ok(aggregate) => aggregate,
+                            Ok(verified_aggregate) => verified_aggregates.push(verified_aggregate),
                             // If we already know the attestation, don't broadcast it or attempt to
                             // further verify it. Return success.
                             //
                             // It's reasonably likely that two different validators produce
                             // identical aggregates, especially if they're using the same beacon
                             // node.
-                            Err(AttnError::AttestationAlreadyKnown(_)) => return Ok(()),
+                            Err(AttnError::AttestationAlreadyKnown(_)) => continue,
                             Err(e) => {
                                 return Err(warp_utils::reject::object_invalid(format!(
```

Comment:
In the name of optimizing validator participation, it might be best to collect the `Err` into an `errors: Vec<_>` and still process the valid attestations.

Then, for the return value, would could serialize all the errors along with the indices to which they pertain.

Reply:
Updated this, let me know what you think. There are some errors that should maybe be 500's instead of 400's, not sure if/how you'd want to differentiate those